<a href="https://colab.research.google.com/github/SnigdhaPanakanti/World_Development_Explorer/blob/main/Part_B_Draft_FInal.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Individual Project PART-B**

**Name: Snigdha Panakanti**

## A Report comparing Food production, Agricultural land and Food insecurities in people.

**INTRODUCTION:**

Agriculture provides the largest share of food supplies and ensures a critical number of ecosystem services all over the world. Though all the countries are rich in producing food, there are also some insecuirities/scarcity of food in the population. This report explains the detail analysis of food production, agricultural land area, food insecurities in people and their undernourishment among the three countries over the period of 8 years (2010-2018).

Indicators:

  - Population in total
  - Food Production index
  - Agricultural Land (% of land area)
  - Prevalence of moderate or severe food insecurity in the population (%)
  - Prevalence of undernourishment (% of population)
  - Permanent Cropland (% of land area)

Time Span: 2010 - 2018
Countries considered: India, Thailand and United States.

In [1]:
import pandas as pd
import plotly
import plotly.express as px

In [2]:
DATA_URL = "https://raw.githubusercontent.com/SnigdhaPanakanti/World_Development_Explorer/main/wdi_data.csv"

In [3]:
df = pd.read_csv(DATA_URL)

df.sample(5)

Unnamed: 0  Year         value       indicator Country Code  \
55           55  2012  9.108000e+01  AG.PRD.FOOD.XD          USA   
133         133  2018  7.900000e+00  SN.ITK.DEFC.ZS          THA   
65           65  2013  6.043946e+01  AG.LND.AGRI.ZS          IND   
93           93  2014  5.262462e+01  AG.LND.ARBL.ZS          IND   
23           23  2015  3.207390e+08     SP.POP.TOTL          USA   

      Country Name               Region         Income Group    Lending Type  
55   United States        North America          High income  Not classified  
133       Thailand  East Asia & Pacific  Upper middle income            IBRD  
65           India           South Asia  Lower middle income            IBRD  
93           India           South Asia  Lower middle income            IBRD  
23   United States        North America          High income  Not classified

**Agricultural Land (% of Land Area):**


- Agricultural Land changes in each country over the span of 2010-2018 is given in the graph below. There is a consistency in the agricultural land area in India throughout the period. United States has a slight decrease in its land percentage during 2011-2013 and then it has increased back to around 40% of the land. Whereas, Thailand have the least area around 10% and increased over the period of 2011-2013 to 40%.


In [4]:
df_agriculture = df.query("Year == 2010").query("indicator == 'AG.LND.AGRI.ZS'")
df_agriculture

Unnamed: 0  Year      value       indicator Country Code   Country Name  \
62          62  2010  60.397418  AG.LND.AGRI.ZS          IND          India   
71          71  2010  41.222181  AG.LND.AGRI.ZS          THA       Thailand   
80          80  2010  44.492513  AG.LND.AGRI.ZS          USA  United States   

                 Region         Income Group    Lending Type  
62           South Asia  Lower middle income            IBRD  
71  East Asia & Pacific  Upper middle income            IBRD  
80        North America          High income  Not classified

In [5]:
df_agri_mean = df_agriculture.groupby('Country Name').mean().reset_index().sort_values(by="value", ascending=False)

df_agri_mean

Country Name  Unnamed: 0    Year      value
0          India        62.0  2010.0  60.397418
2  United States        80.0  2010.0  44.492513
1       Thailand        71.0  2010.0  41.222181

In [ ]:
fig = px.bar(
    df_agri_mean, 
    x="Country Name", 
    labels={"value":"2010 Agricultural land percentage", "Region":""},
    y="value",
    color="Country Name",
    height=800,
    template=list(plotly.io.templates.keys())[5],
    title="Agricultural land (%)"
)

fig.update_layout(showlegend=False)

fig.show()

In [ ]:
fig = px.line(
    df_agri_mean,
    x= "Year",
    labels={"value":"Agricultural land (% of land area)"},
    y= "value",
    height= 700,
    color= "Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "TimeSeries of Agricultural land (%)"
)

fig.show()

**Permanent Crop Land (% of land area):**

- The permanent cropland in the countries during the decade has no much change except for India where the percentage of permanent cropland land has increased slightly.
- The comparision to Agricultural land and permanent cropland is done to check the percentage of cropland for available agricultural land in each country. The median for the agricultural land is 44.49 and for permanent cropland is 4.11. The observed probability factor (P > |t|) is 0.736.

In [8]:
df_perm_cropland = df.query("indicator == 'AG.LND.CROP.ZS'")

df_perm_cropland.sample(5)

Unnamed: 0  Year     value       indicator Country Code   Country Name  \
144         144  2011  4.167241  AG.LND.CROP.ZS          IND          India   
154         154  2012  8.808158  AG.LND.CROP.ZS          THA       Thailand   
143         143  2010  4.111745  AG.LND.CROP.ZS          IND          India   
167         167  2016  0.295165  AG.LND.CROP.ZS          USA  United States   
145         145  2012  4.305140  AG.LND.CROP.ZS          IND          India   

                  Region         Income Group    Lending Type  
144           South Asia  Lower middle income            IBRD  
154  East Asia & Pacific  Upper middle income            IBRD  
143           South Asia  Lower middle income            IBRD  
167        North America          High income  Not classified  
145           South Asia  Lower middle income            IBRD

In [ ]:
fig = px.line(
    df_perm_cropland,
    x= "Year",
    labels={"value":"Permanent Crop land (% of land area)"},
    y= "value",
    height= 700,
    color= "Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "Permanent Crop land (% of land area)"
)

fig.show()

**Food Production Index:**

- Food production is the largest source of environmental degradation. To be sustainable, food production must occur within food-related planetary boundaries for climate change, biodiversity loss, land and water use, as well as for nitrogen and phosphorus cycles.

In [10]:
df_food_production = df.query("indicator == 'AG.PRD.FOOD.XD'")

df_food_production.sample(5)

Unnamed: 0  Year   value       indicator Country Code   Country Name  \
53          53  2010   92.42  AG.PRD.FOOD.XD          USA  United States   
41          41  2016  102.23  AG.PRD.FOOD.XD          IND          India   
43          43  2018  111.77  AG.PRD.FOOD.XD          IND          India   
48          48  2014  105.34  AG.PRD.FOOD.XD          THA       Thailand   
54          54  2011   91.56  AG.PRD.FOOD.XD          USA  United States   

                 Region         Income Group    Lending Type  
53        North America          High income  Not classified  
41           South Asia  Lower middle income            IBRD  
43           South Asia  Lower middle income            IBRD  
48  East Asia & Pacific  Upper middle income            IBRD  
54        North America          High income  Not classified

-  As a brief, the food production in India, Thailand and USA are shown below.

In [ ]:
fig = px.pie(df_food_production,
             values = 'value',
             names = 'Country Name',
             color = 'Country Name',
             template=list(plotly.io.templates.keys())[5],
            )
fig.update_traces(textposition='inside', textinfo='percent+label')
fig.show()

- Over the span of 8 years, the countries show a diverse effect in food production each year. Through a line graph in time series, these differences are shown. Though there are multiple ups and downs in the scale of food production over a period, it has an inclination at the end showing the increase in food production in the countries.

In [ ]:
fig = px.line(
    df_food_production,
    x= "Year",
    labels={"value":"Food Production Index (2010)"},
    y= "value",
    height= 700,
    color= "Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "Food Production Index (2010)"
)

fig.show()

**Food insecurities in the population:**

- Having a diet which is both sufficient in terms of energy (caloric) requirements and diverse to meet additional nutritional needs is essential for good health. Food insecurities among the people is one of the major issues in the world, which inturn leads to the food scarcity. there is a major issue of food insecurities in Thailand compared to India and USA.


In [13]:
df_food_insecurity = df.query("indicator == 'SN.ITK.MSFI.ZS'")

df_food_insecurity

Unnamed: 0  Year  value       indicator Country Code   Country Name  \
27          27  2015   15.1  SN.ITK.MSFI.ZS          THA       Thailand   
28          28  2016   19.7  SN.ITK.MSFI.ZS          THA       Thailand   
29          29  2017   24.8  SN.ITK.MSFI.ZS          THA       Thailand   
30          30  2018   26.4  SN.ITK.MSFI.ZS          THA       Thailand   
31          31  2015   10.5  SN.ITK.MSFI.ZS          USA  United States   
32          32  2016    9.8  SN.ITK.MSFI.ZS          USA  United States   
33          33  2017    8.9  SN.ITK.MSFI.ZS          USA  United States   
34          34  2018    8.4  SN.ITK.MSFI.ZS          USA  United States   

                 Region         Income Group    Lending Type  
27  East Asia & Pacific  Upper middle income            IBRD  
28  East Asia & Pacific  Upper middle income            IBRD  
29  East Asia & Pacific  Upper middle income            IBRD  
30  East Asia & Pacific  Upper middle income            IBRD  
31        North America          High income  Not classified  
32        North America          High income  Not classified  
33        North America          High income  Not classified  
34        North America          High income  Not classified

In [ ]:
fig = px.line(
    df_food_insecurity,
    x= "Year",
    labels={"value":"Food insecurity in the population"},
    y= "value",
    height= 700,
    color= "Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "Moderate or Severe Food insecurity in the population (%)"
)

fig.show()

**Prevalence of undernourishment:**

- Due to the various reasons like food scarcity, lack of nutritious food, etc., people face undernourishment issues. The prevalence of undernourishment is shown below through a line graph. 

In [15]:
df_undernourishment = df.query("indicator == 'SN.ITK.DEFC.ZS'")

df_undernourishment.sample(5)

Unnamed: 0  Year  value       indicator Country Code   Country Name  \
142         142  2018    2.5  SN.ITK.DEFC.ZS          USA  United States   
124         124  2018   14.0  SN.ITK.DEFC.ZS          IND          India   
119         119  2013   15.1  SN.ITK.DEFC.ZS          IND          India   
126         126  2011    9.2  SN.ITK.DEFC.ZS          THA       Thailand   
139         139  2015    2.5  SN.ITK.DEFC.ZS          USA  United States   

                  Region         Income Group    Lending Type  
142        North America          High income  Not classified  
124           South Asia  Lower middle income            IBRD  
119           South Asia  Lower middle income            IBRD  
126  East Asia & Pacific  Upper middle income            IBRD  
139        North America          High income  Not classified

In [ ]:
fig = px.line(
    df_undernourishment,
    x= "Year",
    labels={"value":"Prevalence of undernourishment"},
    y= "value",
    height= 700,
    color= "Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "Prevalence of undernourishment (%)"
)

fig.show()

**Permanent Cropland - Choropleth:**


In [17]:
df_choropleth = df.query("indicator == 'AG.LND.CROP.ZS'")

df_choropleth.sample(5)

Unnamed: 0  Year     value       indicator Country Code   Country Name  \
153         153  2011  8.808158  AG.LND.CROP.ZS          THA       Thailand   
143         143  2010  4.111745  AG.LND.CROP.ZS          IND          India   
166         166  2015  0.295165  AG.LND.CROP.ZS          USA  United States   
144         144  2011  4.167241  AG.LND.CROP.ZS          IND          India   
164         164  2013  0.295165  AG.LND.CROP.ZS          USA  United States   

                  Region         Income Group    Lending Type  
153  East Asia & Pacific  Upper middle income            IBRD  
143           South Asia  Lower middle income            IBRD  
166        North America          High income  Not classified  
144           South Asia  Lower middle income            IBRD  
164        North America          High income  Not classified

In [ ]:
fig = px.choropleth(
    df_choropleth,
    locations= "Country Code",
    labels={"value":"AG.LND.CROP.ZS"},
    color= "value",
    height= 700,
    color_continuous_scale= "geyser",
    hover_name="Country Name",
    template=list(plotly.io.templates.keys())[5],
    title= "2010 Permanent CropLand (% of land area)"
)

fig.show()

**Conclusion:**

- There are various changes in the food production, food nourishment and food insecurity in all the countries. As we have seen the graphs comparing the factors among the countries, there are few changes in the food production over the years. Every country should take care of such concerns like food scarcity/shortage and many more so that people don't die due to hunger.

- It can be determined that the percentage of agricultural land has no bigger changes in India and United States over the period of 8 years. Thailand has an increase in its agricultural land to 40%. The permanent cropland also remained same.